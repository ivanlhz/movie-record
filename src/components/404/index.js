import React from 'react';
import './style.css';

const NotFound = () => (
  <h1>
    404 : <span>Not found</span>
  </h1>
);
export default NotFound;
