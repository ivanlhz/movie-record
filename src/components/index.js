import NotFound from './404';
import MovieDetails from './movie-details';
import MovieCard from './movie-card';
import Search from './search';

export { NotFound, MovieCard, MovieDetails, Search };
