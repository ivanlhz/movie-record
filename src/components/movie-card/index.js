import React from 'react';
import { Link } from 'react-router-dom';
import 'normalize.css';
import './style.css';

const MovieCard = props => (
  <div className="card">
    <Link to={`movies/${props.id}`}>
      <img src={props.poster} alt="" />
    </Link>
  </div>
);
export default MovieCard;
