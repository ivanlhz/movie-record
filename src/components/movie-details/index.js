import React, { Component } from 'react';
import './style.css';
import movies from '../../data/movies.json';

class MovieDetails extends Component {
  constructor(props) {
    super(props);

    this.state = movies.find(movie => movie.id === parseInt(props.match.params.id, 10));
  }

  render() {
    return (
      <div className="movie-details">
        <div className="poster">
          <img className="img-poster" src={this.state.poster} alt={this.state.title} />
        </div>
        <div className="director">{this.state.director} </div>
        <div className="date">{this.state.year}</div>
        <div className="description">
          <h2>{this.state.title}</h2>
          <p>{this.state.plot}</p>
        </div>
        <div className="genre">{this.state.genre}</div>
      </div>
    );
  }
}

export default MovieDetails;
