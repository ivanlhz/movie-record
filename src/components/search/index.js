import React, { Component } from 'react';
import './style.css';
import MovieCard from '../movie-card';
import movies from '../../../data/movies.json';

class Search extends Component {
  state = {
    searchTerm: ''
  };

  handleSearchTerm = event => {
    this.setState({ searchTerm: event.target.value });
  };

  parseJsonFile(jsonFile) {
    return JSON.parse(JSON.stringify(jsonFile));
  }

  getMovies() {
    return this.parseJsonFile(movies)
      .filter(movie => movie.title.toUpperCase().indexOf(this.state.searchTerm.toUpperCase()) >= 0)
      .map(movie => <MovieCard key={movie.id} {...movie} />);
  }

  render() {
    return (
      <div className="search">
        <header>
          <h1>Search</h1>
          <input
            onChange={this.handleSearchTerm}
            value={this.state.searchTerm}
            placeholder="type title to search..."
            type="text"
          />
        </header>
        <div className="movie-list">{this.getMovies()}</div>
      </div>
    );
  }
}

export default Search;
