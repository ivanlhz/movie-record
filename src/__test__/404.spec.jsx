import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import preload from '../../data/movies.json';

import { NotFound } from '../components';

configure({ adapter: new Adapter() });

describe('NotFound', () => {
  it('should be render', () => {
    const component = shallow(<NotFound />);
    expect(component).toMatchSnapshot();
  });
});
