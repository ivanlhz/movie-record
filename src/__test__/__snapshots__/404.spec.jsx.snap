// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should be render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "404 : ",
        <span>
          Not found
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "404 : ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Not found",
        },
        "ref": null,
        "rendered": "Not found",
        "type": "span",
      },
    ],
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "404 : ",
          <span>
            Not found
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "404 : ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Not found",
          },
          "ref": null,
          "rendered": "Not found",
          "type": "span",
        },
      ],
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
